src/PowerShellEditorServices/Hosting/EditorServicesServerFactory.cs

```diff
@@ -2,81 +2,30 @@
 // Licensed under the MIT License.
 
 using System;
-using System.Diagnostics;
 using System.IO;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
-using Microsoft.PowerShell.EditorServices.Logging;
 using Microsoft.PowerShell.EditorServices.Server;
-using Serilog;
-using Serilog.Events;
 using OmniSharp.Extensions.LanguageServer.Protocol.Server;
 using Microsoft.PowerShell.EditorServices.Services.Extension;
 
-#if DEBUG
-using Serilog.Debugging;
-#endif
+// The HostLogger type isn't directly referenced from this assembly, however it uses a common IObservable interface and this alias helps make it more clear the purpose. We can use Microsoft.Extensions.Logging from this point because the ALC should be loaded, but we need to only expose the IObservable to the Hosting assembly so it doesn't try to load MEL before the ALC is ready.
+using HostLogger = System.IObservable<(int logLevel, string message)>;
 
 namespace Microsoft.PowerShell.EditorServices.Hosting
 {
     /// <summary>
-    /// Factory class for hiding dependencies of Editor Services.
+    /// Factory for creating the LSP server and debug server instances.
     /// </summary>
-    /// <remarks>
-    /// Dependency injection and logging are wrapped by factory methods on this class so that the
-    /// host assembly can construct the LSP and debug servers without directly depending on <see
-    /// cref="Microsoft.Extensions.Logging"/> and <see
-    /// cref="Microsoft.Extensions.DependencyInjection"/>.
-    /// </remarks>
     internal sealed class EditorServicesServerFactory : IDisposable
```

Comment:
Does it still need to be an `IDisposable`?

Reply:
Not entirely sure but I was going to clean this up in the logging PR